### DISCUSSION TITLE
'gq` formatting with textwidth

### DISCUSSION THREAD

**Discussion by madjxatw** [LazyVim/LazyVim#534] [General]
I used to use `gq`  (e.g. `gqip`) to format comment block to have all lines wrapped on 80 columns. Since `formatexpr` is set to `vim.lsp.formatexpr()`, the simple`gp` formatting no longer works and I have to use `gw` instead. However, `gw` is remapped in LazyVim. Is there any approach to have `gq` work again with this simple textwidth wrapping case or I have to remove the custom keymap for `gw`?

**Comment by Cruising0904** [comment]
I put vim.cmd("unamp H") to disable it. I am so sure this is right way though.

**Reply by madjxatw** [reply]
> I put vim.cmd("unamp H") to disable it. I am so sure this is right way though.

Did you actually mean `unmap <S-h>`? I think there is nothing to do with `<S-h>`. What I am currently doing is disabling the remap of  `gw` in `lua/configs/keymaps.lua`

**Reply by acut3** [reply]
Thanks @madjxatw, this is just what I was looking for. Your solution of adding `vim.cmd.unmap("gw")` to `lua/config/keymaps.lua` worked great for me.

**Comment by pkazmier** [comment]
Thanks for this as it has been driving me bonkers!

I had no idea that `gw` was just like `gq` with the exception that it uses its own internal formatter versus `formatexpr`/`formatprg`. This is exactly what I've been looking for as I, too, like to format my comments wrapped at 80.

I'm not sure why LazyVim remaps `gw` to search for word under cursor as I've always used the default vim mapping of `*` to do the same. I'll promptly remove that binding so I can have my comment wrapping back again.

@madjxatw Do you also have your comments wrap at `textwidth` as you type them? Any downsides to that? I see the vim default is 0 (so no auto wrapping), but I was thinking of setting it to 80 globally.

**Reply by madjxatw** [reply]
`gw` might be favoured by the author, but I personally think '#' and '*' are sufficient.
I haven't found any downside of wrapping comment along typing. Currently I have no global `textwidth` set but per `filetype` or project.

**Comment by gonzus** [comment]
I arrived here because I like lazyvim, but found that key "s" is mapped to something called "Flash", and does not do text substitution anymore -- I would like to disable this mapping and go back to the original.

Since some of you mentioned "removing keybindings" in this thread, could you share a couple of examples of how to do this, and where to put such change?

**Reply by dpetka2001** [reply]
```lua
return {
  {
    "folke/flash.nvim",
    keys = {
      { "s", mode = { "n", "x", "o" }, false },
    },
  },
}
```
in a file inside `~/.config/nvim/lua/plugins/` directory.

**Reply by gonzus** [reply]
Thank you so much @dpetka2001!

**Reply by madjxatw** [reply]
any other key recommend for plugins like flash, leap, etc instead of the default `s` key? I also prefer to keep `s` as its original behavior as I use it quite often. But I suspect that `s` might be the best tradeoff selected by the plugin authors.

**Comment by llimllib** [comment]
Here's what I ended up doing, just removing `formatexpr` on attach to any buffer because I prefer vim's built-in formatting to LSP's for `gq` usage:

```lua
-- remove the lazyvim mapping for gw, which can be used to format while
-- ignoring formatexpr
-- https://github.com/LazyVim/LazyVim/discussions/534
vim.keymap.del({ "n", "x" }, "gw")
-- Then, reset formatexpr so that gq uses vim's default formatting instead of
-- trying to use the LSP, which I always hate. idea stolen from here, but I
-- don't care at all about keeping gq using formatters
-- https://github.com/willnorris/dotfiles/blob/0e227cef/config/nvim/lua/config/keymaps.lua#L79C1-L89C5
-- See: https://github.com/jose-elias-alvarez/null-ls.nvim/issues/1131
require("lazyvim.util").on_attach(function(_, buf)
  vim.bo[buf].formatexpr = nil
end)
```

**Comment by folke** [comment]
fyi: I just removed the gw keymap from LazyVim
